src/hooks/use-deep-equality.ts

```diff
@@ -1,18 +1,38 @@
 import {
-  useState, useEffect, SetStateAction, Dispatch,
+  Dispatch, SetStateAction, useCallback, useState,
 } from 'react';
 import deepEquals from 'fast-deep-equal';
 
-export default function useDeepEquality<T>(value?: T | undefined): [T | undefined, Dispatch<SetStateAction<T | undefined>>, T | undefined] {
-  const [deep, setDeepValue] = useState(value);
-  const [candidate, setCandidateValue] = useState(value);
+export default function useDeepEquality<T>(initialValue?: T | undefined): [T | undefined, Dispatch<SetStateAction<T | undefined>>] {
```

Comment:
I hadn't noticed this hook before. I feel like it might be a little bit of an anti-pattern, since any time you see a `setValue()` sort of function you tend to assume it works a certain way and this breaks that assumption. It would be clearer in the code I think to just see...
```typescript
if (!deepEqual(newValue, value)) { 
  setValue(newValue);
}
```

Reply:
I'm not sure I agree with the anti-pattern aspect - there are useThrottle, useDebounce, etc... types of hooks, all of which are effectively alterations to the default assumptions provided by useState. 
Personally, I like to encapsulate this type of logic where possible. That said, I do agree that we could just get rid of this hook and use deepEqual since ideally we only have the deep equal need in this particular one-off case. useDeepEqual is testable though... hmm, I don't know. Either way, you've given me something to ponder - I'll think on it some more.

Reply:
It also protects you from forgetting to wrap your setValue() with the deepEquals check - in those cases where you call setValue from multiple places or pass it to a component as a prop. The solution, in that case, would probably be to use a useCallback (which at that point you're back to the same result as having the useDeepEquality hook...

```javascript
const setValueIfNotDeeplyEqual = useCallback((...) => { 
  if (!deepEqual(newValue, value)) { 
    setValue(newValue);
  }
}, [...]);
```

Reply:
Cool, we can leave it for now. It might be nice at some point to make a quick pass through any of those hooks that are wrappers for `useState` and make them `useState____` just for clarity.